.github/workflows/python-package.yaml

```diff
@@ -37,16 +38,15 @@ jobs:
 
       - name: Install python packages
         run: |
-          python -m pip install --upgrade pip wheel
-          pip install --upgrade --upgrade-strategy eager -e .[dev]
+          uv sync --only-dev
```

Comment:
nit: I'm pretty sure we can call `uv build` without running `uv sync` first because the dependencies needed for building are declared in the `build-system` table in `pyproject.toml`

Reply:
true! but i need to install the deps to run `prefect dev build-ui`